src/precompiles/blake2.rs

```diff
@@ -80,8 +80,12 @@ impl Precompile for Blake2F {
         }
         let finished = input[212] != 0;
 
-        let res = blake2::blake2b_f(rounds, h, m, t, finished).to_vec();
-        Ok((ExitSucceed::Returned, res, cost))
+        let output = blake2::blake2b_f(rounds, h, m, t, finished).to_vec();
+        Ok(PrecompileOutput {
+            cost,
+            output,
+            ..Default::default()
```

Comment:
This is cool syntax, I don't think I didn't know you could do this. I'm not sure if I like it though. It means if we add a new field to `PrecompileOutput` in the future and we do not want it to have a default value then the compiler will not help us find the places where the struct initialization needs to be updated.

Maybe a better way to do this would be to define a `new` function on `PrecompileOutput` that only takes `cost` and `output` arguments.

Reply:
I had considered that as I'm aware of the downfall of that. This also has been a bit of topic of hotly debate for use of those as arguments on public methods... For example, you get the benefit of methods being backwards compatible, but that -may- have undesirable consequences. Since this isn't meant to be public, I was putting faith in the team to update all arguments that would use this as it should be pretty straightforward to catch them all. That and I feel quite doubtful that we would add more to it.

Reply:
Discussed with @joshuajbouw privately. We both agree the probability of this causing a problem in the future is low, but also that using a constructor function here is also ok and erring on the side of compile-time safety is preferred for internal structures like this. I'll push a change shortly.

Reply:
Done in https://github.com/aurora-is-near/aurora-engine/pull/116/commits/d7e29374eef5aa5158dab02c09ce5638766b4b8f